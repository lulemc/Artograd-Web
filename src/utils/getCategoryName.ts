import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPages.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPages.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPages.newTender.categories.murals' },
  { id: 3, name: 'tendersPages.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPages.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPages.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPages.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPages.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPages.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPages.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
