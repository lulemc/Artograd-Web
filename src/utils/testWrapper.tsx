import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { RootState, rootReducer, store } from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { createStore } from '@reduxjs/toolkit';
import { initialState as identityState } from '../store/identitySlice';
import { initialState as helpersState } from '../store/helpersSlice';
import { initialState as profileInformationReducer } from '../store/slices/profileInformationSlice';
import { initialState as profileFundrasingReducer } from '../store/slices/profileFundrasingSlice';
import { PersistPartial } from 'redux-persist/lib/persistReducer';

export const initialState: RootState & PersistPartial = {
  identity: identityState,
  helpers: helpersState,
  profileInformation: profileInformationReducer,
  profileFundrasing: profileFundrasingReducer,
  _persist: {
    version: 1,
    rehydrated: true,
  },
};

export const testWrapper = async ({
  component,
  history,
  path,
  state = initialState,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  path?: string;
  state?: RootState;
}) => {
  return path ? (
    <Provider store={store}>
      <MemoryRouter initialEntries={[path]}>{component}</MemoryRouter>
    </Provider>
  ) : (
    renderWithContextAsync(
      <Provider store={createStore(rootReducer, state)}>
        <Router history={history}>{component}</Router>
      </Provider>,
    )
  );
};
