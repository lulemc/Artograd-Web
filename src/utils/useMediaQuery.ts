import { useEffect, useState } from 'react';

export enum DeviceType {
  mobile = '(max-width: 767px)',
  tablet = '(max-width: 1023px)',
  laptop = '(min-width: 1024px)',
}

export const useMediaQuery = (query: DeviceType) => {
  const mediaMatch = window.matchMedia(query);
  const [matches, setMatches] = useState(mediaMatch.matches);
  useEffect(() => {
    const handler = (e: { matches: boolean }) => setMatches(e.matches);
    mediaMatch.addListener(handler);
    return () => mediaMatch.removeListener(handler);
  });
  return matches;
};
