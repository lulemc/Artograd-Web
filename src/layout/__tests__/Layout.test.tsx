import { Layout } from '../Layout';
import { createMemoryHistory } from 'history';
import { testWrapper } from '../../utils/testWrapper';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/test',
  }),
}));

describe('Layout component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', async () => {
    const history = createMemoryHistory();

    const component = await testWrapper({
      component: <Layout>test component</Layout>,
      history,
    });

    expect(component).toMatchSnapshot();
  });
});
