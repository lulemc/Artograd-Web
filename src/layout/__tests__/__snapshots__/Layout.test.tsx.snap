// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; min-height: 100vh;"
      >
        <div
          class=""
          style="min-width: 0; flex-basis: 0px;"
        >
          <nav
            class="cnoL3W uui-main_menu uui-mainmenu-container XHDzBR"
          >
            <div
              class="jR0uTv _3W15HW"
            >
              <div
                class="uui-flex-row XTYWVA iqWEr6"
              >
                <div
                  data-testid="header-logo-image"
                >
                  <a
                    aria-disabled="true"
                    class="uui-button-box uui-disabled -clickable uui-anchor wCHoFl VmWL2z"
                    tabindex="-1"
                  >
                    <img
                      alt="Main Menu Logo"
                      class="gL4W-Y"
                      src="/artograd.logo.svg"
                    />
                  </a>
                </div>
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Home
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Tenders
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Art Objects
                  </div>
                </button>
                <div
                  class="_6SHr2j"
                />
                <div
                  class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                  style="column-gap: 12px;"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH languageSelector"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="uui-caption"
                    >
                      en
                    </div>
                    <div
                      class="S0eIfh uui-icon uui-enabled"
                    >
                      <svg
                        class=""
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m12 16.4-6-6L7.4 9l4.6 4.6L16.6 9l1.4 1.4-6 6z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
                <div
                  class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                  style="column-gap: 12px;"
                >
                  <a
                    class="uui-button-box uui-enabled -clickable uui-anchor QPwdZA JIKmkz uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                    href="https://artograd-dev2.auth.eu-central-1.amazoncognito.com/login?client_id=5c2gu1dvnte5qcvb60rb87llm8&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                    tabindex="0"
                  >
                    <div
                      class="uui-caption"
                    >
                      Sign In
                    </div>
                  </a>
                </div>
                <div
                  class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                  style="column-gap: 12px;"
                >
                  <a
                    class="uui-button-box uui-enabled -clickable uui-anchor QPwdZA JIKmkz uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                    href="https://artograd-dev2.auth.eu-central-1.amazoncognito.com/signup?client_id=5c2gu1dvnte5qcvb60rb87llm8&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                    tabindex="0"
                  >
                    <div
                      class="uui-caption"
                    >
                      Sign Up
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <main
          style="flex-grow: 1;"
        >
          test component
        </main>
        <div
          class=""
          style="min-width: 0; flex-basis: 0px;"
        >
          <nav
            class="cnoL3W uui-main_menu uui-mainmenu-container XHDzBR"
          >
            <div
              class="jR0uTv _3W15HW"
            >
              <div
                class="uui-flex-row XTYWVA iqWEr6"
              >
                <div
                  data-testid="footer-logo-image"
                >
                  <a
                    aria-disabled="true"
                    class="uui-button-box uui-disabled -clickable uui-anchor wCHoFl VmWL2z"
                    tabindex="-1"
                  >
                    <img
                      alt="Main Menu Logo"
                      class="gL4W-Y"
                      src="/artograd.logo.svg"
                    />
                  </a>
                </div>
                <div
                  class="_6SHr2j"
                />
                <span
                  class="uui-button-box uui-enabled l7dxYC _0RZWZH"
                  role="menuitem"
                  tabindex="-1"
                >
                  <div
                    class="uui-caption"
                  >
                    © 1993—2024 EPAM Systems. All Rights Reserved.
                  </div>
                </span>
                <div
                  class="_6SHr2j"
                />
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Meet the team
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Privacy policy
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Cookies policy
                  </div>
                </button>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; min-height: 100vh;"
    >
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cnoL3W uui-main_menu uui-mainmenu-container XHDzBR"
        >
          <div
            class="jR0uTv _3W15HW"
          >
            <div
              class="uui-flex-row XTYWVA iqWEr6"
            >
              <div
                data-testid="header-logo-image"
              >
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor wCHoFl VmWL2z"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="gL4W-Y"
                    src="/artograd.logo.svg"
                  />
                </a>
              </div>
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Home
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Tenders
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Art Objects
                </div>
              </button>
              <div
                class="_6SHr2j"
              />
              <div
                class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                style="column-gap: 12px;"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH languageSelector"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    en
                  </div>
                  <div
                    class="S0eIfh uui-icon uui-enabled"
                  >
                    <svg
                      class=""
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m12 16.4-6-6L7.4 9l4.6 4.6L16.6 9l1.4 1.4-6 6z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
              <div
                class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                style="column-gap: 12px;"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor QPwdZA JIKmkz uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                  href="https://artograd-dev2.auth.eu-central-1.amazoncognito.com/login?client_id=5c2gu1dvnte5qcvb60rb87llm8&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign In
                  </div>
                </a>
              </div>
              <div
                class="uui-flex-row _4IvXA9 HrBGpo xxwsO1 Vd9t75 XTYWVA iqWEr6"
                style="column-gap: 12px;"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor QPwdZA JIKmkz uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                  href="https://artograd-dev2.auth.eu-central-1.amazoncognito.com/signup?client_id=5c2gu1dvnte5qcvb60rb87llm8&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign Up
                  </div>
                </a>
              </div>
            </div>
          </div>
        </nav>
      </div>
      <main
        style="flex-grow: 1;"
      >
        test component
      </main>
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cnoL3W uui-main_menu uui-mainmenu-container XHDzBR"
        >
          <div
            class="jR0uTv _3W15HW"
          >
            <div
              class="uui-flex-row XTYWVA iqWEr6"
            >
              <div
                data-testid="footer-logo-image"
              >
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor wCHoFl VmWL2z"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="gL4W-Y"
                    src="/artograd.logo.svg"
                  />
                </a>
              </div>
              <div
                class="_6SHr2j"
              />
              <span
                class="uui-button-box uui-enabled l7dxYC _0RZWZH"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="uui-caption"
                >
                  © 1993—2024 EPAM Systems. All Rights Reserved.
                </div>
              </span>
              <div
                class="_6SHr2j"
              />
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Meet the team
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Privacy policy
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable l7dxYC _0RZWZH footerLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Cookies policy
                </div>
              </button>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
